ASMDocumentation.md

```diff
@@ -0,0 +1,5841 @@
++        movss   DWORD PTR tv78[ebp], xmm0
++        jmp     SHORT $LN9@Clamp_Colo
++$LN8@Clamp_Colo:
+         movss   xmm0, DWORD PTR _f$1[ebp]
+-        movss   DWORD PTR tv81[ebp], xmm0
+-$LN10@Clamp_Colo:
+-        mov     ecx, DWORD PTR _i$2[ebp]
+-        push    ecx
++        movss   DWORD PTR tv78[ebp], xmm0
++$LN9@Clamp_Colo:
++        mov     eax, DWORD PTR _i$2[ebp]
++        push    eax
+         mov     ecx, DWORD PTR _color$[ebp]
+         call    float & Vector4::operator[](int)             ; Vector4::operator[]
+-        movss   xmm0, DWORD PTR tv81[ebp]
++        movss   xmm0, DWORD PTR tv78[ebp]
+         movss   DWORD PTR [eax], xmm0
+         jmp     $LN2@Clamp_Colo
+ $LN3@Clamp_Colo:
+-        jmp     SHORT $LN1@Clamp_Colo
+-$LN5@Clamp_Colo:
+-        mov     esi, DWORD PTR _color$[ebp]
+-        mov     edx, 1065353216                     ; 3f800000H
+-        mov     edi, DWORD PTR [esi]
+-        mov     ebx, edi
+-        sar     edi, 31                             ; 0000001fH
+-        not     edi
+-        and     edi, ebx
+-        cmp     edi, edx
+-        cmovae  edi, edx
+-        mov     DWORD PTR [esi], edi
+-        mov     edi, DWORD PTR [esi+4]
+-        mov     ebx, edi
+-        sar     edi, 31                             ; 0000001fH
+-        not     edi
+-        and     edi, ebx
+-        cmp     edi, edx
+-        cmovae  edi, edx
+-        mov     DWORD PTR [esi+4], edi
+-        mov     edi, DWORD PTR [esi+8]
+-        mov     ebx, edi
+-        sar     edi, 31                             ; 0000001fH
+-        not     edi
+-        and     edi, ebx
+-        cmp     edi, edx
+-        cmovae  edi, edx
+-        mov     DWORD PTR [esi+8], edi
+-        mov     edi, DWORD PTR [esi+12]
+-        mov     ebx, edi
+-        sar     edi, 31                             ; 0000001fH
+-        not     edi
+-        and     edi, ebx
+-        cmp     edi, edx
+-        cmovae  edi, edx
+-        mov     DWORD PTR [esi+12], edi
+-$LN1@Clamp_Colo:
+-        pop     edi
+-        pop     esi
+-        pop     ebx
+         mov     esp, ebp
+         pop     ebp
+         ret     4
+```
+
+</details>
+
+</details>
+
+---
+
+<details>
+<summary>GeneralsMD/Code/Libraries/Source/WWVegas/WWLib/Except.cpp</summary>
+
+This file contains the following inline assembly blocks:
+
+<details>
+<summary>_purecall</summary>
+
+```c++
+int __cdecl _purecall(void)
+{
+	int return_code = 0;
+
+#ifdef WWDEBUG
+	/*
+	** Use int3 to cause an exception.
+	*/
+	WWDEBUG_SAY(("Pure Virtual Function call. Oh No!\n"));
+	_asm int 0x03;
+#endif	//_DEBUG_ASSERT
+
+	return(return_code);
+}
+```
+
+While I can't generate this in godbolt due to the `WWDEBUG` def, `_asm int 0x03` is simply a `__debugbreak()`:
+
+```c++
+#include <intrin.h>
+
+int __cdecl _purecall(void)
+{
+	int return_code = 0;
+
+#ifdef WWDEBUG
+	/*
+	** Use int3 to cause an exception.
+	*/
+	WWDEBUG_SAY(("Pure Virtual Function call. Oh No!\n"));
+	__debugbreak();
```

Comment:
[here](https://github.com/TheAssemblyArmada/Thyme/blob/ccef1e11c1355c6db577a057c06e7d790f1a0333/deps/baseconfig/src/intrinsics.h#L141) is Thyme's cross CPU (for certain values of CPU) vesrsion of the __debugbreak intrinsic.